tests/unit/lms/validation/_canvas_test.py

```diff
@@ -131,12 +145,90 @@ def test_it_raises_ValidationError_if_the_response_body_isnt_valid_json(
         }
 
     @pytest.fixture(
-        params=[canvas_list_files_response_schema, canvas_public_url_response_schema]
+        params=[
+            canvas_authenticated_users_sections_response_schema,
+            canvas_list_files_response_schema,
+            canvas_public_url_response_schema,
+        ]
     )
     def schema(self, request):
         return request.param()
 
 
+class TestCanvasAuthenticatedUsersSectionsResponseSchema:
+    def test_it_returns_a_list_of_valid_section_dicts(self, schema):
+        parsed_params = schema.parse()
+
+        assert parsed_params == [
+            {"id": 101, "name": "section_name_1"},
+            {"id": 102, "name": "section_name_2"},
+            {"id": 103, "name": "section_name_3"},
+        ]
+
+    @pytest.mark.parametrize(
+        "invalid_body,error_messages",
+        [
+            pytest.param(
+                {},
+                ["Missing data for required field."],
+                id="There must be a sections key",
+            ),
+            pytest.param(
+                {"sections": []},
+                ["Shorter than minimum length 1."],
+                id="There must be at least one section",
+            ),
+            pytest.param(
+                {"sections": "not a list"},
+                ["Not a valid list."],
+                id="Sections must be a list",
+            ),
+            pytest.param(
+                {"sections": ["not a dict", 1, False]},
+                {
+                    0: {"_schema": ["Invalid input type."]},
+                    1: {"_schema": ["Invalid input type."]},
+                    2: {"_schema": ["Invalid input type."]},
+                },
+                id="Each section must be a dict",
+            ),
+            pytest.param(
+                {"sections": [{"name": "Test Section"}]},
+                {0: {"id": ["Missing data for required field."]}},
+                id="Each section dict must have an ID",
+            ),
+            pytest.param(
+                {"sections": [{"id": "not an int", "name": "Test Section"}]},
+                {0: {"id": ["Not a valid integer."]}},
+                id="IDs must be ints",
+            ),
+            pytest.param(
+                {"sections": [{"id": 101}]},
+                {0: {"name": ["Missing data for required field."]}},
+                id="Each section dict must have a name",
+            ),
+            pytest.param(
+                {"sections": [{"id": 101, "name": 26}]},
+                {0: {"name": ["Not a valid string."]}},
+                id="Names must be strings",
+            ),
+        ],
+    )
+    def test_invalid(self, schema, invalid_body, error_messages):
+        schema.context["response"].json.return_value = invalid_body
+
+        with pytest.raises(ValidationError) as exc_info:
+            schema.parse()
+
+        assert exc_info.value.messages == {
+            "sections": error_messages,
+        }
```

Comment:
Only about six lines of test code are needed to test `CanvasAuthenticatedUsersSectionsResponseSchema` (counting the lines in this test method plus the one other test method above), and then there are about 50 lines of declarative pytest fixtures covering all the possible failure cases. I think this is fine: there's not much actual test _code_ here, and I'm fine with throwing a large number of cases at the schema. Each of the test cases tests something about the `CanvasAuthenticatedUsersSectionsResponseSchema` class code itself, each test case could be broken by a change to `CanvasAuthenticatedUsersSectionsResponseSchema`